pandas/core/generic.py

```diff
@@ -7988,9 +7985,9 @@ def interpolate(
                 FutureWarning,
                 stacklevel=find_stack_level(),
             )
-        elif np.any(obj.dtypes == object):
+        elif np.any(self.dtypes == object):
```

Comment:
i think this actually needs to be the maybe-transposed object for the interpolate check to be correct.

Reply:
> i think this actually needs to be the maybe-transposed object for the interpolate check to be correct.

I reorganized the code a bit so as to fix this check. I moved one of the deprecations warnings down, but I think it doesn't matter.